utils/prometheus/parser_test.go

```diff
@@ -0,0 +1,247 @@
+    separator: ;
+    regex: "true"
+    replacement: $1
+    action: keep
+  - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scrape_slow]
+    separator: ;
+    regex: "true"
+    replacement: $1
+    action: drop
+- job_name: kubernetes-service-endpoints-slow
+  honor_timestamps: true
+  scrape_interval: 5m
+  scrape_timeout: 30s
+  metrics_path: /metrics
+  scheme: http
+  kubernetes_sd_configs:
+  - role: endpoints
+  relabel_configs:
+  - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_scrape_slow]
+    separator: ;
+    regex: "true"
+    replacement: $1
+    action: keep
+- job_name: prometheus-pushgateway
+  honor_labels: true
+  honor_timestamps: true
+  scrape_interval: 1m
+  scrape_timeout: 10s
+  metrics_path: /metrics
+  scheme: http
+  kubernetes_sd_configs:
+  - role: service
+  relabel_configs:
+  - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_probe]
+    separator: ;
+    regex: pushgateway
+    replacement: $1
+    action: keep
+- job_name: kubernetes-services
+  honor_timestamps: true
+  params:
+    module:
+    - http_2xx
+  scrape_interval: 1m
+  scrape_timeout: 10s
+  metrics_path: /probe
+  scheme: http
+  kubernetes_sd_configs:
+  - role: service
+  relabel_configs:
+  - source_labels: [__meta_kubernetes_service_annotation_prometheus_io_probe]
+    separator: ;
+    regex: "true"
+    replacement: $1
+    action: keep
+- job_name: kubernetes-pods
+  honor_timestamps: true
+  scrape_interval: 1m
+  scrape_timeout: 10s
+  metrics_path: /metrics
+  scheme: http
+  kubernetes_sd_configs:
+  - role: pod
+  relabel_configs:
+  - source_labels: [__meta_kubernetes_pod_phase]
+    separator: ;
+    regex: Pending|Succeeded|Failed|Completed
+    replacement: $1
+    action: drop
+- job_name: kubernetes-pods-slow
+  honor_timestamps: true
+  scrape_interval: 5m
+  scrape_timeout: 30s
+  metrics_path: /metrics
+  scheme: http
+  kubernetes_sd_configs:
+  - role: pod
+  relabel_configs:
+  - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scrape_slow]
+    separator: ;
+    regex: "true"
+    replacement: $1
+    action: keep
+  - source_labels: [__meta_kubernetes_pod_annotation_prometheus_io_scheme]
+    separator: ;
+    regex: (https?)
+    target_label: __scheme__
+    replacement: $1
+    action: replace
+- job_name: carts-sockshop-production
+  honor_timestamps: false
+  scrape_interval: 5s
+  scrape_timeout: 3s
+  metrics_path: /metrics
+  scheme: http
+  static_configs:
+  - targets:
+    - carts.sockshop-production:80
+`
+
+func TestLoad(t *testing.T) {
+	_, err := LoadYamlConfiguration(sampleConfigurationYAML)
+	require.NoError(t, err)
+}
+
+func TestToString(t *testing.T) {
+	config, err := LoadYamlConfiguration(sampleConfigurationYAML)
+
+	require.NoError(t, err)
+	require.NotNil(t, config)
+
+	str := config.String()
+
+	require.NotContainsf(t, str, "<error creating", "yaml marshaling should not contain error string")
+}
```

Comment:
It would be nice to see also that we are able to parse and write back the configuration with and without changes: 

1. load and dump configuration and assert that the 2 files are yaml-equivalent
2. load, modify and dump configuration and assert that the new values are there + the rest is yaml-equivalent